src/components/LoginDialog.jsx

```diff
@@ -1,33 +1,48 @@
 import React from 'react';
-import Button from '@material-ui/core/Button';
 import Dialog from '@material-ui/core/Dialog';
 import DialogContent from '@material-ui/core/DialogContent';
 import DialogContentText from '@material-ui/core/DialogContentText';
 import DialogTitle from '@material-ui/core/DialogTitle';
 
-import FirebaseLoginScreen from './FirebaseLoginScreen';
+import firebase from 'firebase';
+import StyleFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
+import PropTypes from 'prop-types';
 
-export default function LoginDialog() {
-  const [open, setOpen] = React.useState(false);
-  const handleClickOpen = () => setOpen(true);
-  const handleClose = () => setOpen(false);
+class LoginDialog extends React.Component {
+  uiConfig = {
+    signInFlow: 'popup',
+    signInOptions: [
+      firebase.auth.GoogleAuthProvider.PROVIDER_ID,
+      firebase.auth.EmailAuthProvider.PROVIDER_ID,
+      firebase.auth.FacebookAuthProvider.PROVIDER_ID,
+    ],
+    credentialHelper: 'none', // hacky way to disable redirect on email login
+    callbacks: {
+      signInSuccessWithAuthResult: () => false,
+    },
+  }
 
-  return (
-    <div>
-      <Button variant="outlined" color="secondary" onClick={handleClickOpen}>
-        Login
-      </Button>
-      <Dialog open={open} onClose={handleClose} aria-labelledby="responsive-dialog-title">
+  render() {
+    const { open, onClose } = this.props;
+    return (
+      <Dialog open={open} onClose={onClose} aria-labelledby="responsive-dialog-title">
         <DialogTitle id="responsive-dialog-title">Login using your account or email.</DialogTitle>
         <DialogContent>
           <DialogContentText>
             You can login using your existing account to authenticate and log in to
             our web-app. You can create an account if you do not want to use your
             existing account to interact with our app.
           </DialogContentText>
-          <FirebaseLoginScreen />
+          <StyleFirebaseAuth uiConfig={this.uiConfig} firebaseAuth={firebase.auth()} />
         </DialogContent>
       </Dialog>
-    </div>
-  );
+    );
+  }
 }
+
+LoginDialog.propTypes = {
+  open: PropTypes.element.isRequired,
+  onClose: PropTypes.element.isRequired,
```

Comment:
Based on the [PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes) page these should be of type `PropTypes.func`

Reply:
Doing it that way causes lint errors

Reply:
What lint errors does it give?

Reply:
44:3  error  propType "open" is not required, but has no corresponding defaultProps declaration     react/require-default-props
  45:3  error  propType "onClose" is not required, but has no corresponding defaultProps declaration  react/require-default-props

Reply:
Instead of using element as the PropType, use func
`open: PropTypes.func.isRequired` 
`onClose: PropTypes.func.isRequired`

Reply:
Got it and no lint error, thanks Edwin